vscode-lean4/src/leanclient.ts

```diff
@@ -378,9 +377,19 @@ export class LeanClient implements Disposable {
 
     isSameWorkspace(uri: Uri){
         if (this.folderUri) {
-            if (uri.toString().startsWith(this.folderUri.toString())) {
-                // skip it, this file belongs to a different workspace...
-                return true;
+            if (this.folderUri.scheme !== uri.scheme) return false;
+            if (this.folderUri.scheme === 'file') {
+                const realPath1 = fs.realpathSync(this.folderUri.fsPath).toLowerCase();
+                const realPath2 = fs.realpathSync(uri.fsPath).toLowerCase();
```

Comment:
Lower-casing should only be performed on windows.  (And probably on macos as well?)

Reply:
Fixed